src/sdl2_compat.c

```diff
@@ -5374,7 +5371,7 @@ DECLSPEC SDL_Window * SDLCALL
 SDL_CreateWindow(const char *title, int x, int y, int w, int h, Uint32 flags)
 {
     SDL_Window *window = NULL;
-    SDL_bool hidden = (flags & SDL_WINDOW_HIDDEN) ? SDL_TRUE : SDL_FALSE;
+    const SDL_bool hidden = flags & SDL_WINDOW_HIDDEN;
```

Comment:
`(flags & SDL_WINDOW_HIDDEN) != 0`
Remember, this is actually supposed to be a 0 or 1 value, not 0 or non-zero.

Reply:
It's only locally used, but changed as suggested